### PR TITLE
Validate release filename and header on upload

### PR DESCRIPTION
**Pull request by HazAT**
Related to #4743 #4742

### CODE REVIEW

File: src/sentry/api/endpoints/release_files.py [hunk 1]
```diff
@@ -148,6 +149,9 @@ def post(self, request, project, version):
         if not full_name:
             return Response({'detail': 'File name must be specified'}, status=400)
         name = full_name.rsplit('/', 1)[-1]
+        pattern = re.compile(r"(\n|\t|\r|\f|\v|\\n|\\t|\\r|\\f|\\v)")
```
Comment (mitsuhiko) [inline src/sentry/api/endpoints/release_files.py]: You can shorten this to `[\n\t\r\f\v\\]`. This rejects all those whitespaces plus backslashes which also make no sense there.

**Review by mitsuhiko — CHANGES_REQUESTED** [review]

**Comment by HazAT** [comment]
Will change the squash commit message ...

File: src/sentry/api/endpoints/release_files.py [hunk 2]
```diff
@@ -148,6 +149,9 @@ def post(self, request, project, version):
         if not full_name:
             return Response({'detail': 'File name must be specified'}, status=400)
         name = full_name.rsplit('/', 1)[-1]
+        pattern = re.compile(r"[\n\t\r\f\v\\]")
```
Comment (mitsuhiko) [inline src/sentry/api/endpoints/release_files.py]: can you compile the re as a global? like _fn_re = re.compile(...) on module level and then use it from here. The regex compilation in Python is really not very fast.

**Review by mitsuhiko — CHANGES_REQUESTED** [review]

File: src/sentry/api/endpoints/release_files.py [hunk 3]
```diff
@@ -147,8 +149,12 @@ def post(self, request, project, version):
         full_name = request.DATA.get('name', fileobj.name)
         if not full_name:
             return Response({'detail': 'File name must be specified'}, status=400)
+
         name = full_name.rsplit('/', 1)[-1]
 
+        if re.search(_filename_re, name):
```
Comment (mitsuhiko) [inline src/sentry/api/endpoints/release_files.py]: Canonical way is to use `_filename_re.search(name)` directly.

**Review by mitsuhiko — CHANGES_REQUESTED** [review]

**Review by mitsuhiko — APPROVED** [review]

**Comment by wrt2** [comment]
creative commons- atribution-sharealike 3 0-cc by-sa3 0

**Comment by mitsuhiko** [comment]
Note: I deleted a comment that was here because the account was suspicious (only random activity but the activity that was there referred to a cryptolocker software)
